connectors/kafka/src/main/java/forklift/connectors/KafkaConnector.java

```diff
@@ -0,0 +1,135 @@
+package forklift.connectors;
+
+import forklift.consumer.ForkliftConsumerI;
+import forklift.consumer.KafkaTopicConsumer;
+import forklift.controller.KafkaController;
+import forklift.message.MessageStream;
+import forklift.producers.ForkliftProducerI;
+import forklift.producers.KafkaForkliftProducer;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Manages both consuming and producing events on the kafka message broker.
+ */
+public class KafkaConnector implements ForkliftConnectorI {
+    private static final Logger log = LoggerFactory.getLogger(KafkaConnector.class);
+    private final String kafkaHosts;
+    private final String schemaRegistries;
+    private final String groupId;
+    private KafkaProducer<?, ?> kafkaProducer;
+    private MessageStream messageStream = new MessageStream();
+    private KafkaController controller;
+
+    /**
+     * Constructs a nw instance of the KafkaConnector
+     *
+     * @param kafkaHosts       list of kafka servers in host:port,... format
+     * @param schemaRegistries list of schema registry servers in http://host:port,... format
+     * @param groupId          the groupId to use when subscribing to topics
+     */
+    public KafkaConnector(String kafkaHosts, String schemaRegistries, String groupId) {
+        this.kafkaHosts = kafkaHosts;
+        this.schemaRegistries = schemaRegistries;
+        this.groupId = groupId;
+    }
+
+    @Override
+    public void start() throws ConnectorException {
+        //We do nothing here.  Consumer and producer are created when needed
+    }
+
+    private KafkaProducer createKafkaProducer() {
+
+        Properties producerProperties = new Properties();
+        producerProperties.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaHosts);
+        producerProperties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+                               io.confluent.kafka.serializers.KafkaAvroSerializer.class);
+        producerProperties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+                               io.confluent.kafka.serializers.KafkaAvroSerializer.class);
+        //schema.registry.url is a comma separated list of urls
+        producerProperties.put("schema.registry.url", schemaRegistries);
+        kafkaProducer = new KafkaProducer(producerProperties);
+        return kafkaProducer;
+    }
+
+    private KafkaController createController() {
+        Properties props = new Properties();
+        props.put("bootstrap.servers", kafkaHosts);
```

Comment:
Seems weird to use string property names here and the constants for the producer.